src/Orckestra.Composer.Search/Services/BaseSearchViewService.cs

```diff
@@ -214,65 +202,22 @@ private IList<SearchSortBy> GetSearchSortByList(SearchType searchType)
-            productSearchVm.ProductId = productDocument.ProductId;
-            productSearchVm.BrandId = ExtractLookupId("Brand_Facet", productDocument.PropertyBag);
-            MapProductSearchViewModelInfos(productSearchVm, productDocument, cultureInfo);
-            MapProductSearchViewModelUrl(productSearchVm, variantId, cultureInfo, createSearchViewModelParam.SearchParam.Criteria.BaseUrl);
-            MapProductSearchViewModelImage(productSearchVm, imgDictionary);
-            productSearchVm.IsAvailableToSell = await GetProductSearchViewModelAvailableForSell(productSearchVm, productDocument).ConfigureAwait(false);
-            var pricing = await PriceProvider.GetPriceAsync(productSearchVm.HasVariants, productDocument).ConfigureAwait(false);
-            MapProductSearchViewModelPricing(productSearchVm, pricing);
-            productSearchVm.IsRecurringOrderEligible = RecurringOrdersSettings.Enabled && productDocument.PropertyBag.IsRecurringOrderEligible();
-
-            productSearchVm.Context["IsRecurringOrderEligible "] = productSearchVm.IsRecurringOrderEligible;
-
-            return productSearchVm;
-        }
-
-        protected virtual string ExtractLookupId(string fieldName, PropertyBag propertyBag)
-        {
-            if (propertyBag == null) { return null; }
-            var fieldValue = propertyBag.ContainsKey(fieldName) 
-                ? (propertyBag[fieldName] as string ?? (propertyBag[fieldName] as string[])[0])
-                : null;
-            if (string.IsNullOrWhiteSpace(fieldValue)) { return null; }
-
-            var extractedValues = fieldValue.Split(new[] { "::" }, StringSplitOptions.None);
-            return extractedValues.Length < 3
-                ? null
-                : extractedValues[2];
-        }
 
-        protected virtual async Task<bool> GetProductSearchViewModelAvailableForSell(
-            ProductSearchViewModel productSearchViewModel,
-            ProductDocument productDocument)
+        // NOTE: when fetching data for products from OCC APIs, make sure to query data in batches for optimal performance
+        // https://docs.orckestra.com/developer-documentation/platform-performance/batch-api-requests
+        protected virtual async Task<IList<ProductSearchViewModel>> AppendProductSearchViewModels(IList<(ProductSearchViewModel, ProductDocument)> productSearchResultList)
         {
-            if (productSearchViewModel.HasVariants) { return true; }
-
             _productSettings = await ProductSettings.GetProductSettings(ComposerContext.Scope, ComposerContext.CultureInfo).ConfigureAwait(false);
 
-            if (!_productSettings.IsInventoryEnabled) { return true; }
+            foreach (var (productSearchVm, productDocument) in productSearchResultList)
+            {
+                if (!_productSettings.IsInventoryEnabled) { productSearchVm.IsAvailableToSell = true; }
```

Comment:
I do not like that IsAvailableToSell  is calculated in both places: in Factory and here.
Probalbly _productSettings can be a parameter for the Factory and can be used to create model inside Factory.
Also please request @napernik to review this request.
Thanks

Reply:
Yes, currently IsAvailableToSell is set in 2 places, when one needs settings, and the other one doesn't.
Most likely the MapProductSearchViewModelAvailableForSell would also need to use settings, as it reads the available statuses from **ComposerConfiguration.AvailableStatusForSell** is used instead of reading it from _productSettings, as in OCC it has a field **AvailableInventoryStatuses**.

I would suggest to set the status only in MapProductSearchViewModelAvailableForSell, and for it to receive ProductSettings as a parameter.

Whether to call it from Factory, or from AppendProductSearchViewModels - it is probably equally good